### PR TITLE
Feat/process nuclei results

### LINKED ISSUE #227: New scanner: Nuclei
CVE scanner

### PR DESCRIPTION
**Pull request by mohamed-cds**
Process nuclei reports and view the results

Closes #227

### CODE REVIEW

File: api/front_end/templates/scan_results_nuclei.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                      <div class="text-sm font-medium text-gray-900">
+                        <p class="flex flex-wrap gap-2">
+                          <span class="{{ 'bg-red-300' if security_report.critical | default(0) > 0 else 'bg-gray-100'}} inline-block p-1 px-2">{{security_report.summary['critical'] or 0 }}&nbsp;Critical</span>
+                          <span class="{{ 'bg-yellow-600' if security_report.summary.high | default(0) > 0 else 'bg-gray-100'}} inline-block p-1 px-2">{{security_report.summary['high'] or 0 }}&nbsp;High</span>
+                          <span class="{{ 'bg-yellow-300' if security_report.summary.medium | default(0) > 0 else 'bg-gray-100'}} inline-block p-1 px-2">{{security_report.summary['medium'] or 0 }}&nbsp;Medium</span>
+                          <span class="{{ 'bg-green-300' if security_report.summary.low | default(0) > 0 else 'bg-gray-100'}} inline-block p-1 px-2">{{security_report.summary['low'] or 0 }}&nbsp;Low</span>
+                          <span class="{{ 'bg-blue-300' if security_report.summary.info | default(0) > 0 else 'bg-gray-100'}} inline-block p-1 px-2">{{security_report.summary['info'] or 0 }}&nbsp;Informational</span>
```
Comment (patheard) [inline api/front_end/templates/scan_results_nuclei.html:64]: Should the severity text at the end of these lines be localized?

File: api/front_end/templates/scan_results_nuclei_summary.html [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                            {% for security_violation in report.security_violations %}
+                                <tr>
+                                    <td class="p-5">
+                                        <div id="violation{{ loop.index }}" class="flex align-baseline mb-5">
+                                            <a class="mr-4 text-2xl" href="https://dequeuniversity.com/rules/axe/4.1/{{ security_violation.violation }}" target="_blank">
+                                                {{ security_violation.violation }}
+                                            </a>
+                                            <span class="theme-impact p-2 py-1 whitespace-nowrap capitalize" data-impact="{{ security_violation.impact }}">{{ security_violation.impact }}</span>
+                                        </div>
+                                        <table class="mb-4 min-w-full border border-gray-200">
```
Comment (patheard) [inline api/front_end/templates/scan_results_nuclei_summary.html:38]: Do we need a nested `<table>` for this layout?

File: api/front_end/templates/scan_results_owasp_zap.html [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                      </div>
+                    </td>
+                    <td class="px-6 py-4 whitespace-nowrap">
+                      <div class="text-sm font-medium text-gray-900">
+                        <p class="flex flex-wrap gap-2">
+                          {% set reduced_summary = security_report.summary|extract_risk_text %}
+                          <span class="{{ 'bg-red-300' if reduced_summary.high | default(0) > 0 else 'bg-gray-100'}} inline-block p-1 px-2">{{reduced_summary['high'] or 0 }}&nbsp;High</span>
+                          <span class="{{ 'bg-yellow-300' if reduced_summary.medium | default(0) > 0 else 'bg-gray-100'}} inline-block p-1 px-2">{{reduced_summary['medium'] or 0 }}&nbsp;Medium</span>
+                          <span class="{{ 'bg-green-300' if reduced_summary.low | default(0) > 0 else 'bg-gray-100'}} inline-block p-1 px-2">{{reduced_summary['low'] or 0 }}&nbsp;Low</span>
+                          <span class="{{ 'bg-blue-300' if reduced_summary.informational | default(0) > 0 else 'bg-gray-100'}} inline-block p-1 px-2">{{reduced_summary['informational'] or 0 }}&nbsp;Info</span>
```
Comment (patheard) [inline api/front_end/templates/scan_results_owasp_zap.html:64]: Same comment around localizing severity strings.

**Review by patheard — COMMENTED** [review]
Had a few questions/comments.

File: api/front_end/templates/scan_results_nuclei_summary.html (hunk 2 above)
Comment (mohamed-cds) [reply api/front_end/templates/scan_results_nuclei_summary.html:38]: copy pasted from the a11y summary view. I think this should be tackled outside this PR as we refactor the views to look more like the existing a11y tracker.

**Review by patheard — APPROVED** [review]

**Pull request merged**